kong/plugins/request-transformer/access.lua

```diff
@@ -212,15 +212,24 @@ local function transform_headers(conf)
 
   -- Add header(s)
   for _, name, value in iter(conf.add.headers) do
-    name = name:lower()
-    if not headers[name] and name ~= HOST then
+    if not headers[name] and name:lower() ~= HOST then
       headers[name] = value
     end
   end
 
   -- Append header(s)
   for _, name, value in iter(conf.append.headers) do
     if name:lower() ~= HOST then
+      if headers[name] ~= nil then -- keep original content, use configd case
+        for header_name, _ in pairs(headers) do
+          if name:lower() == header_name:lower() then
+            local content = headers[header_name]
+            headers[header_name] = nil
+            headers[name] = content
+            break
+          end
+        end
+      end
```

Comment:
Do we need this change? From the descriptions of https://github.com/Kong/kong-plugin-request-transformer/issues/28 it seems that only `add.headers` has the issue and `append.headers` does not lower case the headers anyway.

Reply:
We can leave this change out, but my approach here was to keep the same behavior that we'd get with the `add.headers`, i.e. force the configured case to be used.

Reply:
See comment below.